sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala

```diff
@@ -168,7 +168,7 @@ object GpuDeviceManager extends Logging {
     // Align workaround for https://github.com/rapidsai/rmm/issues/527
     def truncateToAlignment(x: Long): Long = x & ~511L
 
-    var initialAllocation = truncateToAlignment((conf.rmmAllocFraction * info.total).toLong)
+    var initialAllocation = truncateToAlignment((conf.rmmAllocFraction * info.free).toLong)
     if (initialAllocation > info.free) {
```

Comment:
This check is useless now, but we probably want to have some checks for a minimum amount of free memory and warn about that, even before the initial allocation.